File: frontend/src/services/feedback.service.js
import {api} from "../api.js";

export const createFeedback = async (data) => {
    try {
        const response = await api.post("/feedbacks", data);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const getAllFeedbacks = async () => {
    try {
        const response = await api.get("/feedbacks");
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const getFeedbackById = async (feedbackId) => {
    try {
        const response = await api.get(`/feedbacks/${feedbackId}`);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const updateFeedback = async (feedBackId, data) => {
    try {
        const response = await api.put(`/feedbacks/${feedBackId}`, data);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const deleteFeedback = async (feedBackId) => {
    try {
        const response = await api.delete(`/feedbacks/${feedBackId}`);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const addFeedbackResponse = async (feedbackId, data) => {
    try {
        const response = await api.post(`/feedbacks/${feedbackId}/response`, data);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const getUserFeedbacks = async () => {
    try {
        const response = await api.get("/feedbacks/me");
        return response.data;
    }
    catch (e) {
        console.error(e);
    }
}

File: feedback backend/src/main/java/com/sliit/crave/service/FeedBackService.java
package com.sliit.crave.service;


import com.sliit.crave.dto.request.feedback.FeedBackResponseRequest;
import com.sliit.crave.dto.request.feedback.FeedbackRequest;
import com.sliit.crave.dto.response.Response;
import com.sliit.crave.dto.response.feedback.FeedbackDTO;
import com.sliit.crave.dto.response.feedback.FeedbackResponseDTO;

import java.util.List;

public interface FeedBackService {
    Response<FeedbackDTO> createFeedback(FeedbackRequest feedbackDto);
    Response<List<FeedbackDTO>> getAllFeedBacks();
    Response<FeedbackDTO> getFeedBackById(Long feedBackId);
    Response<FeedbackDTO> updateFeedback(Long feedBackId, FeedbackRequest feedbackDto);
    Response<Void> deleteFeedback(Long feedBackId);
    Response<FeedbackResponseDTO> addResponse(Long feedbackId, FeedBackResponseRequest request);
    Response<List<FeedbackDTO>> getFeedBacksByUserId(Long userId);

}
